Classes/Command/UpdateInheritanceCommand.php

```diff
@@ -0,0 +1,301 @@
+                    $queryBuilder->createNamedParameter(0, \PDO::PARAM_INT)
+                )
+            )
+            ->orderBy('tstamp', 'DESC')
+            ->execute()
+            ->fetchAllAssociative();
+
+        return $records ?? [];
+    }
+
+    /**
+     * Rebuild inheritance queue and return number of products added to queue.
+     *
+     * @return int
+     */
+    protected function rebuildInheritanceQueue(): int
+    {
+        $this->emptyInheritanceQueue();
+
+        $allChilds = $this->fetchChildProducts();
+        foreach ($allChilds as $index => $child) {
+            $this->addProductToInheritanceQueue($child['uid']);
+        }
+
+        return count($allChilds);
+    }
+
+    /**
+     * Fetch number if products in inheritance queue.
+     *
+     * @return int
+     */
+    protected function fetchInheritanceQueueCount(): int
+    {
+        /** @var \TYPO3\CMS\Core\Database\Query\QueryBuilder $queryBuilder */
+        $queryBuilder = GeneralUtility::makeInstance(ConnectionPool::class)
+            ->getQueryBuilderForTable(InheritanceQueueRepository::TABLE_NAME);
+        $queryBuilder->getRestrictions()->removeAll();
+
+        return $queryBuilder
+            ->addSelectLiteral('COUNT(*) as cnt')
+            ->from(InheritanceQueueRepository::TABLE_NAME)
+            ->where(
+                $queryBuilder->expr()->eq(
+                    'indexed',
+                    $queryBuilder->createNamedParameter(0, \PDO::PARAM_INT)
+                ),
+                $queryBuilder->expr()->eq(
+                    'errors',
+                    $queryBuilder->createNamedParameter('', \PDO::PARAM_STR)
+                )
+            )
+            ->execute()
+            ->fetchOne();
+    }
+
+    /**
+     * Fetch products to update inheritance for.
+     *
+     * @return int
+     */
+    protected function fetchInheritanceQueue(int $limit): array
+    {
+        /** @var \TYPO3\CMS\Core\Database\Query\QueryBuilder $queryBuilder */
+        $queryBuilder = GeneralUtility::makeInstance(ConnectionPool::class)
+            ->getQueryBuilderForTable(InheritanceQueueRepository::TABLE_NAME);
+        $queryBuilder->getRestrictions()->removeAll();
+
+        return $queryBuilder
+            ->addSelectLiteral('product_uid')
+            ->from(InheritanceQueueRepository::TABLE_NAME)
+            ->where(
+                $queryBuilder->expr()->eq(
+                    'indexed',
+                    $queryBuilder->createNamedParameter(0, \PDO::PARAM_INT)
+                ),
+                $queryBuilder->expr()->eq(
+                    'errors',
+                    $queryBuilder->createNamedParameter('', \PDO::PARAM_STR)
+                )
+            )
+            ->setMaxResults($limit)
+            ->execute()
+            ->fetchAllAssociative();
+    }
+
+    /**
+     * Empty inheritance queue.
+     *
+     * @return void
+     */
+    protected function emptyInheritanceQueue(): void
```

Comment:
Maybe better with truncate than empty? I found myself thinking of it as a boolean return, but that could be hasEmpty… as well, so I don't know.

```suggestion
    protected function truncateInheritanceQueue(): void
```